### ISSUE TITLE
Help with results

### ISSUE THREAD

**Issue by Branikas** [milesial/Pytorch-UNet#81]
Hi everyone, thanks for your work Milesial, it is really appreciated.
I have some questions I would like to ask.
I am trying to train the model on some data and I have noticed that for the first epochs the validation error starts decreasing and after some point it just oscillates around a value.
I suspect a few things, first it could be because of the Adam optimizer, it happens by its nature and also I noticed when I try to train the model again from the last checkpoint and the optimization starts from fresh the validation accuracy increases for the first epochs.
Another thing is that while the dice coeff score is around 0.82 half of the pixels are classified wrong. Do you think it is beacuse of the dice coeff? My image has only a few pixels belonging to the positive class so because dice coeff gives weight to the true positive maybe I should try the good old cross entropy?
Thank you in advance!

**Comment by milesial** [comment]
Hi, it may be that because Adam is an adaptative optimizer, when you load from a checkpoint the adapted learning rates are reset, so you should try smaller learning rates when loading from checkpoints.
If your validation loss oscillate around a value, it may be near a local optimum, so try lowering the learning rate.
You could change the validation metric if you wish, but that won't affect the training itself as cross entropy is used for the training anyway.

**Comment by Branikas** [comment]
Thanks for the fast response @milesial! Yes, i know Adam is adaptive that's why I asked. I will try with a smaller lr once i get a GPU, i m running on CPU for the moment. I will aso try to train on a switching optimizer, the process starts with adam and then switches to SGD with momentum when some criteria are met, I ll try to work on implementing that.
Thank you so much for all the help!

**Comment by milesial** [comment]
Yes, go ahead and experiment, that's the best way to learn ! Good luck :)
